Porting/updateAUTHORS.pl

```diff
@@ -94,72 +96,178 @@ sub main {
+            "Is '$self->{$_}' up to date?")
+            for qw(authors_file mailmap_file exclude_file);
+
+        ok(!$self->{missing_author}{$_},
+            sprintf "%s is listed in AUTHORS", _clean_name($_))
+            for sort keys %{$self->{missing_author}||{}};
+
+        my $uncommitted_changes= $self->uncommitted_changes;
+        if ($uncommitted_changes) {
+            my ($author_name, $author_email)= $self->current_author_name_email();
+            my ($committer_name, $committer_email)= $self->current_committer_name_email();
+
+            ok($author_name && $author_email,"git knows your author name and email.");
+            ok($committer_name && $committer_email,"git knows your author name and email.");
+
+            my $author_known= $self->known_contributor($author_name,$author_email);
+            my $committer_known= $self->known_contributor($committer_name,$committer_email);
+            if (is( $author_known && $committer_known, 1,
+                "Uncommitted changes are by a known contributor?"))
+            {
+                diag "Testing uncommtted changes! Remember to commit before you push!";
+            } else {
+                diag uncommitted_changes($author_name,$author_email,$committer_name, $committer_email);
+            }
+        } else {
+            ok(!$uncommitted_changes,"No changes on disk");
+        }
+
+        diag "\nFiles need updating! ",
+             "You probably just need to run\n\n",
+             "   Porting/updateAUTHORS.pl\n\n",
+             "and commit the results."
+             if $self->changed_count;
+        done_testing();
+        return 0;
     }
-    if ($self->{show_rank}) {
+    elsif ($self->{show_rank}) {
         $self->report_stats("who_stats", "author");
+        return 0;
     }
     elsif ($self->{show_applied}) {
         $self->report_stats("who_stats", "applied");
+        return 0;
     }
     elsif ($self->{show_stats}) {
         my @fields= ("author", "applied", "committer");
         push @fields,
             ("num_files", "lines_added", "lines_removed", "lines_delta")
             if $self->{numstat};
         $self->report_stats("who_stats", @fields);
+        return 0;
     }
     elsif ($self->{show_files}) {
         $self->report_stats(
             "file_stats",  "commits", "lines_added", "lines_removed",
             "lines_delta", "binary_change"
         );
+        return 0;
     }
     elsif ($self->{show_file_changes}) {
         $self->report_stats(
             "file_stats", "lines_delta", "lines_added", "lines_removed",
             "commits"
         );
+        return 0;
     }
     elsif ($self->{show_file_chainsaw}) {
         $self->{in_reverse}= !$self->{in_reverse};
         $self->report_stats(
             "file_stats", "lines_delta", "lines_added", "lines_removed",
             "commits"
         );
+        return 0;
     }
     elsif ($self->{show_who}) {
         $self->print_who();
+        return 0;
```

Comment:
i added the returns because each action is supposed to do one thing and return, and the if statement was getting long.  I didnt /intend/ for the final return to be 1, but its a good idea. :-)